### ISSUE TITLE
Add check for UnverifiedCovenantError

### ISSUE THREAD

**Issue by dagurval** [Bitcoin-com/cashscript#56]
While trying to narrow down an issue with items left on the stack, I stumbled on a different issue:

```
$ ts-node bug.ts 
Contract address: bitcoincash:pzq7x3fpxgus4dcwsyu8mxh6gcttxncf3y2qjakgsu
Contract balance: 20000
contract opcount: undefined
contract bytesize: undefined
{ TfTypeError: Expected property "3" of type UInt32, got Number -1
    at captureStackTrace (/home/dagurval/cashscript/node_modules/typeforce/errors.js:20:11)
    at tfSubError (/home/dagurval/cashscript/node_modules/typeforce/errors.js:99:3)
    at /home/dagurval/cashscript/node_modules/typeforce/index.js:190:17
    at Array.every (<anonymous>)
    at _tuple (/home/dagurval/cashscript/node_modules/typeforce/index.js:186:20)
    at typeforce (/home/dagurval/cashscript/node_modules/typeforce/index.js:233:9)
    at Transaction.sigHashPreimageBufChunks (/home/dagurval/cashscript/node_modules/cramer-bch/src/transaction.js:321:3)
    at Transaction.sigHashPreimageBuf (/home/dagurval/cashscript/node_modules/cramer-bch/src/transaction.js:416:29)
    at inputs.forEach (/home/dagurval/cashscript/examples/node_modules/cashscript/src/Transaction.ts:154:22)
    at Array.forEach (<anonymous>)
  message: 'Expected property "3" of type UInt32, got Number -1',
  __label: undefined,
  __property: 3,
  __type: { [Function: UInt32] toJSON: [Function: bound ] },
  __value: -1,
  __valueTypeName: 'Number' }
transaction details: undefined
```

Code: https://github.com/dagurval/cashscript/commit/53ff2018f488c07f21b49f2b41521af534b6bf80

**Comment by rkalis** [comment]
Thanks for the bug report. Will check it out.

**Comment by rkalis** [comment]
I found the problem. It's a really stupid bug in the compiler 😅

So when  you are using covenant fields (like `tx.hashOutputs`) you need to "verify" the covenant by including a `require(checkSig(..., ...));`.  The compiler has a traversal to ensure that this is included, otherwise it should throw a `UnverifiedCovenantError`. The code for this is in place and is properly unit tested. But then I forgot to actually activate the functionality in the `compile` function. So it doesn't actually check this and compiles even though the contract is invalid. This creates problems down the line because the CashScript SDK expects a correct contract. I will put a fix for this in the next release.

**Comment by dagurval** [comment]
So it was a user error :-). Thanks for looking into it so quickly. After adding `checkSig`, I managed to reproduce the issue with stack elements and create an issue for it #57

**Comment by rkalis** [comment]
This is released in v0.3.2
